ramlapi.go

```diff
@@ -3,73 +3,40 @@ package ramlapi
 import (
 	"fmt"
 	"log"
+	"os"
 
 	"github.com/buddhamagnet/raml"
 )
 
-// ProcessRAML processes a RAML file and returns an API definition.
-func ProcessRAML(ramlFile string) (*raml.APIDefinition, error) {
-	routes, err := raml.ParseFile(ramlFile)
-	if err != nil {
-		return nil, fmt.Errorf("Failed parsing RAML file: %s\n", err.Error())
+// Build takes a RAML API definition, a router and a routing map,
+// and wires them all together.
+func Build(api *raml.APIDefinition, fun func(data map[string]string)) {
+	for name, res := range api.Resources {
+		processResource("", name, &res, fun)
 	}
-	return routes, nil
 }
 
-// processResource recursively process resources and their nested children
-// and returns the path so far for the children. The function takes a routerFunc
-// as an argument that is invoked with the verb, resource path and handler as
-// the resources are processed, so the calling code can use pat, mux, httprouter
-// or whatever router they desire and we don't need to know about it.
-func processResource(parent, name string, resource *raml.Resource, routerFunc func(data map[string]string)) string {
-
-	var resourcepath = parent + name
-	log.Println("processing", name, "resource")
-	log.Println("path: ", resourcepath)
-
-	for verb, details := range ResourceVerbs(resource) {
-		log.Println("--- " + verb)
-		data := map[string]string{
-			"verb":    verb,
-			"path":    resourcepath,
-			"handler": details["handler"],
-		}
-		if details["query"] != "" {
-			data["query"] = details["query"]
-			data["query_type"] = details["query_type"]
-			data["query_description"] = details["query_description"]
-			data["query_example"] = details["query_example"]
-			data["query_pattern"] = details["query_pattern"]
-		}
-		routerFunc(data)
-	}
-
-	// Get all children.
-	for nestname, nested := range resource.Nested {
-		return processResource(resourcepath, nestname, nested, routerFunc)
+// Process processes a RAML file and returns an API definition.
+func Process(file string) (*raml.APIDefinition, error) {
+	routes, err := raml.ParseFile(file)
+	if err != nil {
+		return nil, fmt.Errorf("Failed parsing RAML file: %s\n", err.Error())
```

Comment:
Don't need to call Error(). The function will see the error interface is being implemented.